### PR TITLE
Replaces qpid electron with vcabbage/amqp

### PR DESCRIPTION
**Pull request by csibbitt**
* Hoping for a performance increase
  * electron added concurrency abstractions to proton with global locking
  * proton was a C-binding to golang
  * new library is native golang
* We did indeed get a performance increase!
  * Can now push ~25k/s instead of ~16k/s 
  * Now when I ask for 10k/s I get EXACTLY 10k/s instead of 8905/s
* Disabled message limit testing for now
  * We don't use it
  * If this performs WAY better I might re-enable it
* Re-tested all major functionality including threading and Acks
  * MAY have fixed the ack related memory leak?
* Removed qpid libraries from build
* Added a quick "firehose" sanity test tool I happened to find useful

### CODE REVIEW

File: cmd/telemetry-bench.go [hunk 1]
```diff
@@ -288,13 +290,8 @@ func getMessagesLimit(urls string, metricsInAmqp int, enableCPUProfile bool) {
...
 	}
 	os.Exit(0)
-	/*
-		close(cancelMesg)
-		close(cancel)
-		waitb.Wait()
-		con.Close(nil)
-	*/
 }
+*/
```
Comment (csibbitt) [inline cmd/telemetry-bench.go]: See L353 below

**Comment by csibbitt** [comment]
@atyronesmith FYI; this library seems promising!

**Comment by csibbitt** [comment]
[Try using vcabbage/amqp in telemetry-bench](https://trello.com/c/vqyduO1j/718-try-using-vcabbage-amqp-in-telemetry-bench)

**Comment by csibbitt** [comment]
```
[csibbitt@laptop telemetry-bench (csibbitt-718-vcabbage) ]$ .git/hooks/pre-commit
Formatting, Linting, and Vetting all returned OK.
```

**Comment by paramite** [comment]
If you are experiencing such performance improvement, maybe we should also to the same switch in smart-gateway also. As additional positive thing which this switch will bring, is avoiding dependency on qpid-proton libs.

**Comment by leifmadsen** [comment]
> If you are experiencing such performance improvement, maybe we should also to the same switch in smart-gateway also. As additional positive thing which this switch will bring, is avoiding dependency on qpid-proton libs.
> 
> ... and then I read the trello card :D

Ha! :)

Yea telemetry-bench first to prove things out, then SG next :)

**Pull request merged**